services/user.go

```diff
@@ -0,0 +1,43 @@
+package services
+
+import (
+	"github.com/EloYaniel/academy-go-q42021/apiclient"
+	e "github.com/EloYaniel/academy-go-q42021/entities"
+	repo "github.com/EloYaniel/academy-go-q42021/repositories/contracts"
+)
+
+type UserService struct {
+	repo      repo.UserRepository
+	apiClient apiclient.ApiClient
+}
+
+func NewUserService(repo repo.UserRepository, client apiclient.ApiClient) *UserService {
+	return &UserService{repo: repo, apiClient: client}
+}
+
+func (s *UserService) GetUsers() ([]e.User, error) {
+	users, _ := s.repo.GetUsers()
+
+	if len(users) == 0 {
+		resp := struct {
+			Data []e.User
+		}{}
+		err := s.apiClient.Get("https://reqres.in/api/users", nil, &resp)
```

Comment:
Just like how you added the file path as an attribute to your repo struct, you can add this url as well there, we don't really want to have this kind of info hardcoded.